src/Serializer/SerializerContextBuilder.php

```diff
@@ -94,7 +94,7 @@ public function createFromRequest(Request $request, bool $normalization, array $
         }
 
         foreach ($resourceMetadata->getItemOperations() as $operation) {
-            if ('PATCH' === $operation['method'] && \in_array('application/merge-patch+json', $operation['input_formats']['json'] ?? [], true)) {
+            if (\in_array('application/merge-patch+json', $operation['input_formats']['json'] ?? [], true) && 'PATCH' === $operation['method']) {
```

Comment:
```suggestion
            if ('PATCH' === $operation['method'] && \in_array('application/merge-patch+json', $operation['input_formats']['json'] ?? [], true)) {
```

Because the comparison is faster.